src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/issuenav/IssueNavResultsView.kt

```diff
@@ -0,0 +1,26 @@
+package com.atlassian.performance.tools.jiraactions.api.page.issuenav
+
+import com.atlassian.performance.seleniumjs.NativeExpectedCondition
+
+interface IssueNavResultsView {
```

Comment:
Do we want this interface to be our SPI (in API)? An alternative could be an selection type similar to `enum`, so that we could hide the methods of this interface from API. 

My assumption is that we don't want the consumer to implement their own versions of this interface. Probably Jira has limited number of different search views, so I guess the lib has the implementation list complete. If that's the case we don't need to expose the messaging between implementations of `IssueNavResultsView` and `SearchIssues`.

On the other hand I'm actually a fan of designing software that exposes everything that it does instead of one that has a lot of room to do hidden stuff to achieve outcome. I personally believe that the consumer should have all the power to define composition layer of the application and rebuild the dependencies to work (or not work) as they want. Default composition should be enough to limit configuration complexity.

Reply:
Hiding it would be nice. But it cannot be an `enum`, because:
* [enums block evolution](https://github.com/wyrzyk/java-api-design-checklist/blob/master/CHECKLIST.md#255-do-not-add-new-values-to-a-released-enumeration)
* [they get unwieldy very fast](https://github.com/atlassian/jira-actions/commit/6988b29451c9a840ccee8591ca481d6b513d5ec2#diff-d611ddd88e4934022a3e1e93fdd8d14e28636cab88bf1f8b04f76d2fa08f4a71L93-R104) (and this is before more methods like `listIssueKeys`
* there are more variations of interacting with the views than we think, e.g. we might detect them via the switcher icon rather than via the results

Reply:
> enums block evolution

yup, I know, that's why I used "selection type similar to `enum`" wording 😉 

I wonder if the restriction around `enum` makes us unable to pass choices in non-polymorphic way or is there some other good way to do that. Not sure if passing `int` would be a good idea for use case like that - feels a bit cheap.